SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_SizeChecker.cpp

```diff
@@ -0,0 +1,264 @@
+            console.log("Detected overworld.");
+            return;
+        case 1:
+            console.log("Detected dialog box.");
+            pbf_press_button(context, BUTTON_A, 20, 5);
+            continue;
+
+        default:
+            throw OperationFailedException(
+                ErrorReport::SEND_ERROR_REPORT, console,
+                "exit_check_mode(): No recognized state after 60 seconds.",
+                true
+            );
+        }
+
+    }
+
+}
+
+
+
+void SizeChecker::program(SingleSwitchProgramEnvironment& env, BotBaseContext& context){
+    assert_16_9_720p_min(env.logger(), env.console);
+
+    SizeChecker_Descriptor::Stats& stats = env.current_stats<SizeChecker_Descriptor::Stats>();
+
+    //  Connect the controller.
+    pbf_press_button(context, BUTTON_LCLICK, 10, 0);
+
+    // Loop through boxes.
+    for (uint8_t box = 0; box < BOXES_TO_CHECK; box++){
+        enter_check_mode(env.console, context);
+        // Default state on box entry does not highlight selection, move cursor so it does.
+        pbf_press_dpad(context, DPAD_LEFT, 5, 100);
+        context.wait_for_all_requests();
+
+        if (box > 0){
+            move_to_right_box(context);
+        }
+        context.wait_for_all_requests();
+
+        // Loop through the rows and columns.
+        for (uint8_t row = 0; row < 5; row++){
+            for (uint8_t col = 0; col < 6; col++){
+
+                move_box_cursor(env.program_info(), env.console, context, BoxCursorLocation::SLOTS, row, col);
+
+                SomethingInBoxSlotDetector sth_in_box_detector(COLOR_RED);
+                BoxCurrentEggDetector egg_detector;
+                VideoOverlaySet overlays(env.console.overlay());
+                sth_in_box_detector.make_overlays(overlays);
+                egg_detector.make_overlays(overlays);
+                context.wait_for_all_requests();
+
+                VideoSnapshot screen = env.console.video().snapshot();
+
+                if (!sth_in_box_detector.detect(screen)){
+                    env.console.log("Detected empty cell.");
+                    stats.m_empty++;
+                    env.update_stats();
+                    continue;
+                }
+
+                if (egg_detector.detect(screen)){
+                    env.console.log("Detected egg in cell.");
+                    stats.m_eggs++;
+                    env.update_stats();
+                    continue;
+                }
+
+                // Initiate size checking prompt.
+                pbf_press_button(context, BUTTON_A, 20, 20);
+
+                // If last cell not empty.
+                if (row == 4 && col == 5){
+                    exit_check_mode(env.console, context);
+                } else {
+                    enter_check_mode(env.console, context);
+                    // Default state on box entry does not highlight selection, move cursor so it does.
+                    pbf_press_dpad(context, DPAD_LEFT, 5, 100);
+                }

```

Comment:
I don't understand this part, do you mind explaining ?

Reply:
I agree, this is too hacky. I've updated the overall logic to be more intuitive now so it should make more sense.